rules/manage-javascript-projects-with-nx/rule.md

```diff
@@ -0,0 +1,57 @@
+---
+type: rule
+archivedreason: 
+title: Tools - Do you use Nx to manage your huge JavaScript solution?
+guid: da45f247-e727-4f83-8a89-89cf2a407b31
+uri: manage-javascript-projects-with-nx
+created: 2023-02-10T01:26:45.262Z
+authors:
+- title: Chris Clement
+  url: https://www.ssw.com.au/people/chris-clement
+related: []
+
+---
+
+One of the main problems working on a huge monorepo solution is usually the development experience and the build time.
+[Nx](https://nx.dev/) is one of many tools that can improve this experience in JavaScript projects.
+
+<!--endintro-->
+
+The amount of code that needs to be processed by the compiler scales proportionally with the solution size. Hence, the compile time will grow naturally as the solution grows in size.
+This surely affects both the development experience and the team's velocity, leaving both developers and stakeholders unhappy.
+
+[Nx](https://nx.dev/) is a JavaScript build system that aims to make developing on monorepo solution easier and faster.
+Nx offers the following features:
+- [Cached build](https://nx.dev/concepts/how-caching-works) - faster development and build time
+- [Task Pipeline](https://nx.dev/concepts/task-pipeline-configuration) - provide tools to control how the monorepo build is performed
+- [Dependency graph](https://nx.dev/core-features/explore-graph) - see the relationship between projects
+- [Affected graph](https://nx.dev/concepts/affected) - see which projects is affected by your commit
+- and many more...
+
+Currently, Nx supports many frameworks, such as Angular, React, Node, and [many more](https://nx.dev/packages).
+
+Adding a tool such as Nx to a project will obviously add another moving parts to the solution, so it's a good idea to know the advantages and disadvantages of Nx.
+
+::: good
+Advantages:
+:::
+
+* Faster development build time
+* Faster CI time with [NxCloud](https://nx.app/)
+* Monorepo collaboration tool with [Package based](https://nx.dev/getting-started/package-based-repo-tutorial) or [Integrated repo based](https://nx.dev/getting-started/integrated-repo-tutorial) strategy
+
+::: bad
+Disadvantages:
+:::
+
+* Additional external dependency to be maintained
+* Learning curve
+* Only supports JavaScript projects
+
+
+Consider using Nx in a project when your solution:
+* is a JavaScript monorepo
```

Comment:
@christoment 

Please start this list items with Capitalized words

Reply:
Done :)